cdxev/validator/validate.py

```diff
@@ -22,16 +22,25 @@ def validate_sbom(
     file: Path,
     report_format: t.Optional[str],
     report_path: t.Optional[Path],
-    schema_type: str = "default",
-    filename_regex: t.Optional[str] = "",
-    schema_path: str = "",
+    schema_type: t.Optional[str],
+    filename_regex: t.Optional[str],
+    schema_path: t.Optional[Path],
 ) -> int:
     errors = []
-    if input_format == "json":
-        sbom_schema, used_schema_path = open_schema(
-            sbom, file, schema_type, schema_path
+    if (schema_path is not None) == bool(schema_type):
+        raise AssertionError(
+            "Exactly one of schema_path or schema_type must be non-None"
```

Comment:
Is this even reachable? Now that you removed the default, when both are not `None`, argparse should catch that `schema-type` and `schema-path` are mutually exclusive + if it is not the case, then a test for these lines is missing

Reply:
No, it's not supposed to be reachable. That's why it's an `AssertionError` and not a pretty error message for the user. It would be a programming bug on the side of the caller if that error was ever raised.
But I admit that in different places I dealt with similar issues in different ways. For instance, same file, L161 there is simply a comment stating that assertion in natural language, no actual test at runtime.
I'm open to removing the check if you prefer.

Reply:
If it's all the same to you, I'll leave it, if only to safe a minute of work-